src/components/Switchers/index.js

```diff
@@ -1,30 +1,62 @@
-import React from 'react'
-import styled from 'styled-components'
+import React, { Component, PropTypes } from 'react'
 
-const StyledInput = styled.input`
-  &:checked + span {
-    transform: translateX(100%);
-  }
-`
-
-const StaticSwitcher = () => (
-  <div className='w5 relative flex items-center br-pill ba b--light-navy bg-dark-navy'>
-    <label className='w-50 normal lh-copy tc light-gray relative z-1 pv2 pointer'
-           htmlFor='some-id-left'
-           style={{boxSizing: 'border-box'}}>Monthly</label>
-    <input type='radio' name='some-name' id='some-id-left' className='absolute o-0' />
-    <label className='w-50 normal lh-copy tc light-gray relative z-1 pv2 pointer'
-           htmlFor='some-id-right'
-           style={{boxSizing: 'border-box'}}>Weekly</label>
-    <StyledInput type='radio' name='some-name' id='some-id-right' className='absolute o-0' />
-    <span className='w-50 db br-pill pa2 bg-blue absolute left-0'
+const labelClasses = 'w-50 normal lh-copy tc light-gray relative z-1 pv2 pointer'
+
+const Highlight = ({selectedItem}) => {
+  return (
+    <span className='w-50 db br-pill pa2 bg-blue absolute left-0 border-box'
           style={{
-            boxSizing: 'border-box',
-            transition: '150ms'
+            transition: '150ms',
+            transform: selectedItem === 'right' ? 'translateX(100%)' : ''
           }}>
-      <span className='db h2'></span>
+      <span className='db h2' />
     </span>
-  </div>
-)
+  )
+}
+Highlight.propTypes = {
+  selectedItem: PropTypes.oneOf(['left', 'right']),
+}
+
+class Switcher extends Component {
+  static propTypes = {
+    leftOption: PropTypes.string,
+    rightOption: PropTypes.string,
+    selectedItem: PropTypes.oneOf(['left', 'right']),
+    onClick: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = { selectedItem: this.props.selectedItem === 'left' ? 'left' : 'right' }
+  }
```

Comment:
Can use property initializers to replace the need for a constructor; so instead of lines 28-32, could be:
```
state = {
  selectedItem: this.props.selectedItem === 'left' ? 'left' : 'right'
}
```
Example: https://github.com/eggheadio/egghead-instructor-center/blob/master/src/components/LessonList/index.js#L19-L21